modules/fx/arrow-fx-kotlinx-coroutines/src/main/kotlin/arrow/integrations/kotlinx/extensions.kt

```diff
@@ -0,0 +1,37 @@
+package arrow.integrations.kotlinx
+
+import arrow.core.Either
+import arrow.fx.IOOf
+import arrow.fx.OnCancel
+import arrow.fx.fix
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.newCoroutineContext
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+suspend fun <A> IOOf<A>.suspendCancellable() = suspendCancellableCoroutine<A> { cont ->
+  val disposable = fix().unsafeRunAsyncCancellable(OnCancel.ThrowCancellationException) { result ->
+    result.fold(cont::resumeWithException) { cont.resume(it) }
```

Comment:
`result.fold(cont::resumeWithException, cont::resume)`

Reply:
Yep, already went through that one but the type system is failing here, had to specify the unit explicitly otherwise it won't compile 🤷‍♂ 

> Type inference failed: inline fun <C> fold(ifLeft: (Throwable) -> C, ifRight: (A) -> C): C
cannot be applied to
(KFunction1<@ParameterName Throwable, Unit>,KFunction1<@ParameterName A, Unit>)
FAILURE: Build failed with an exception.

Reply:
This is a conflict where they a `resume` with an additional param which has a default value. Due to that, it can't be used as a lambda reference.

Same issue as we have with the `dummy: Unit = Unit` overloads.